packages/admin/src/form/Autocomplete.tsx

```diff
@@ -0,0 +1,57 @@
+import { CircularProgress, TextField } from "@material-ui/core";
+import Autocomplete, { AutocompleteProps, AutocompleteRenderInputParams } from "@material-ui/lab/Autocomplete";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+
+interface Props<T> extends FieldRenderProps<T, HTMLInputElement | HTMLTextAreaElement> {
+    optionValue?: keyof T;
+    optionLabel?: keyof T;
+}
+
+export const FinalFormAutocomplete = <
+    T extends Record<string, any>,
+    Multiple extends boolean | undefined,
+    DisableClearable extends boolean | undefined,
+    FreeSolo extends boolean | undefined,
+>({
+    input: { onChange, value, ...restInput },
+    optionLabel = "label",
+    optionValue = "value",
+    loading = false,
+    isAsync = false,
+    ...rest
+}: Props<T> & Omit<AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>, "renderInput">) => {
+    return (
+        <Autocomplete
+            getOptionSelected={(option: T, value: T) => {
+                if (!value) return false;
+                return optionValue ? option[optionValue] === value[optionValue] : option === value;
+            }}
+            getOptionLabel={(option: T) => {
+                return optionLabel && option[optionLabel] !== undefined ? option[optionLabel].toString() : "--unknown--";
```

Comment:
can "--unknown--" be rendered in normal circumstances?

if yes, we need to make in translatable.
if no, we should throw an Error instead

Reply:
Only when the optionLabel-prop is set wrong. I think it would be better just to throw an error. I will change this.

Reply:
Now an error is thrown